benchmarks/README.adoc

```diff
@@ -1,55 +1,318 @@
+{'PC': '10092', 'Encoding': '0141', 'WoE': 16, 'Instruction': 'c.addi', 'Category': 'alu_imm', 'Destination': ('sp',), 'Source': ('sp',), 'Immediate': '16'}
+{'PC': '10094', 'Encoding': 'a2b1', 'WoE': 16, 'Instruction': 'c.j', 'Category': 'jump', 'HOB': True, 'Target_address': 66016}
+----
+
+After populating the instruction record, the script finds all branch targets and mark them with a special key to help identify them as (HOB) head of blocks.
+
+The script uses this instruction record to perform various analysis tasks, these tasks are called actions.
+
+Generally speaking, most of these actions correspond to a single Zce instruction. When the script is called with a given action, it would filter the instruction record, find the target instruction/instruction chain, make sure it fits the target optimised instruction, then it replaces that instruction with the optimised instruction. Along side instructions modelling actions, the script contain several actions used to report various statistics in CSV format.
+
+== CMD line arguments 
+HCA -a action[1] action[2] action[n] -f file[1] file[2] file[n] -v[1/2] -r [output-dir]
+
+_Description_: +
+-a Required action names, full description of available actions is available below +
+-f ELF files paths +
+-v Verbosity level : 0  (default value, suppress all warning and debug messages), 1 (prints all errors and important messages), 2 (print debug information along with error messages). +
+-r output directory for CSV results files, if not specific results would be printed to console output.
+
+NOTE: The script is capable of performing any number of actions on any number of files. For each passed file, the  script will perform the required actions as described below, and then write the result as a new row to the the results CSV file or to the console output before moving to the next file.
+
+
+== Available actions:
+
+Below is a list of various actions that the script can execute, please note that actions are case sensitive, and the script expects them exactly like they are written in the headings.
+
+
+=== Zce modelling actions:
+
+==== c_sb,c_sh, c_lhu, c_lbu, c_lh and c_lb
+
+_Description_: These actions estimate the code size savings of c.sb,c.sh,c.lhu, c.lbu, c.lh and c.lb respectively
+
+_Replacement criteria_:
+
+. Find all SB / SH/ LHU/ LBU / SH / LHU instructions
+. If the immediate field bigger than 0 and less than 2^4, and the register fits within compressed register format then replace it with compressed form.
+
+==== beqi and bnei
+
+_Description_: These actions estimate the code size savings of beqi and bnei respectively
+
+_Replacement criteria_:
+
+. Find all beq / bne instructions
+. If one of the operands comes from li, and its value bigger than 0 and less than 2^5 then replace the sequence with either beqi/bnei.
+
+NOTE: We don't need to check if the offset would fit the 
```

Comment:
offset is the same, this line should be removed

Reply:
> offset is the same, this line should be removed

Yes its, I removed it in my local copy, but this copy was sent for approval about a month ago, and I needed to push the exact same copy !